hilti/toolchain/src/compiler/visitors/printer.cc

```diff
@@ -821,8 +821,17 @@ struct Visitor : visitor::PreOrder<void, Visitor> {
         else
             out << fmt("%d..%d", n.lower(), n.upper());
 
-        if ( n.attributes() )
-            out << ' ' << *n.attributes();
+        if ( n.attributes() ) {
+            std::stringstream ss;
+            printer::Stream p(ss, out.isCompact());
+            p << ' ' << (*n.attributes());
+
+            out << ' '
+                << rt::escapeUTF8(ss.str(),
+                                  /*escape_quotes=*/true,
+                                  /*escape_control=*/false,
+                                  /*keep_hex=*/false);
+        }
```

Comment:
This feels a bit like a hack. I think the escaping should be done where the attributes values are printed.

Also, there's not test for the escaping actually kicking in, is there?